### ISSUE TITLE
Helm charts don't work out of the box

### ISSUE THREAD

**Issue by farcaller** [coreos/prometheus-operator#443]
**What did you do?**
```
$ kubectl create ns kube-prometheus
$ helm install --namespace kube-prometheus -n prometheus-operator ./prometheus-operator
...
$ helm install --namespace kube-prometheus -n prometheus ./prometheus
...
# whoops, out of scheduleable mem
$ helm upgrade prometheus ./prometheus --set resources.requests.memory=400Mi
...
$ helm install --namespace kube-prometheus -n alertmanager ./alertmanager
...
$ helm install --namespace kube-prometheus -n kube-prometheus ./kube-prometheus
Error: found in requirements.yaml, but missing in charts/ directory: alertmanager, prometheus
# I commented those two out, they are, supposedly installed
$ helm install --namespace kube-prometheus -n kube-prometheus ./kube-prometheus
...
```

**What did you expect to see?**

Prometheus working at http://localhost:8001/api/v1/proxy/namespaces/kube-prometheus/services/prometheus-prometheus:http

**What did you see instead? Under which circumstances?**

Prometheus is up and running but has no alerts or services being monitored, despite them being installed:

```
$ kubectl get servicemonitors --all-namespaces
NAMESPACE         NAME                                               KIND
kube-prometheus   kube-prometheus-exporter-kube-api                  ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kube-controller-manager   ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kube-dns                  ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kube-etcd                 ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kube-scheduler            ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kube-state                ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kubelets                  ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-kubernetes                ServiceMonitor.v1alpha1.monitoring.coreos.com
kube-prometheus   kube-prometheus-exporter-node                      ServiceMonitor.v1alpha1.monitoring.coreos.com
```

**Environment**

* Kubernetes version information:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.6", GitCommit:"7fa1c1756d8bc963f1a389f4a6937dc71f08ada2", GitTreeState:"clean", BuildDate:"2017-06-16T20:46:19Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.6", GitCommit:"7fa1c1756d8bc963f1a389f4a6937dc71f08ada2", GitTreeState:"clean", BuildDate:"2017-06-16T18:21:54Z", GoVersion:"go1.7.6", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

kubeadm

* Manifests:

this repo's helm packages at a9376416daa446afcba50f16e82de86f63e5f833

* Prometheus Operator Logs:

```
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=kube-prometheus/prometheus
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:579 component=prometheusoperator msg="update handler" old=42490 cur=42491
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:316 component=alertmanageroperator msg="update handler" old=42490 cur=42491
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=kube-prometheus/prometheus
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:316 component=alertmanageroperator msg="update handler" old=42491 cur=42492
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:579 component=prometheusoperator msg="update handler" old=42491 cur=42492
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=kube-prometheus/prometheus
prometheus-operator-prometheus-operator-2899541066-cq23b prometheus-operator ts=2017-06-22T22:39:13Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
...
```

**Comment by brancz** [comment]
@mgoodness ^

**Comment by farcaller** [comment]
I poked around and I think I managed to get *some* of those to work. I guess my problem was that I installed prometheus as a dedicated helm app, and that screwed up some of the configuration.

I added the following overrides:

```yaml
serviceMonitorsSelector:
  app: prometheus
  prometheus: kube-prometheus
rulesSelector:
  app: prometheus
  prometheus: kube-prometheus
```

to make prometheus able to find the relevant rules and service monitors as provisioned by kube-prometheus app. After fixing some rbac issues, I'm only left with a few problems:

kubelet target group is broken with `server returned HTTP status 403 Forbidden`

ServiceMonitor is defined as:
```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: ServiceMonitor
metadata:
  creationTimestamp: 2017-06-22T22:28:43Z
  labels:
    app: prometheus
    chart: exporter-kubelets-0.1.0
    component: kubelets
    heritage: Tiller
    prometheus: kube-prometheus
    release: kube-prometheus
  name: kube-prometheus-exporter-kubelets
  namespace: kube-prometheus
  resourceVersion: "41247"
  selfLink: /apis/monitoring.coreos.com/v1alpha1/namespaces/kube-prometheus/servicemonitors/kube-prometheus-exporter-kubelets
  uid: 264de7bd-579a-11e7-8ee6-525400f18cb3
spec:
  endpoints:
  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
    interval: 15s
    port: https-metrics
    scheme: https
    tlsConfig:
      caFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
      insecureSkipVerify: true
  jobLabel: kube-prometheus-exporter-kubelets
  namespaceSelector:
    matchNames:
    - kube-system
  selector:
    matchLabels:
      k8s-app: kubelet
```

via helm.

Some services don't work either, e.g. etc monitoring targets are missing, despite being defined:

```yaml
# kubectl -n kube-system get svc/kube-prometheus-exporter-kube-etcd -oyaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: 2017-06-22T22:28:43Z
  labels:
    app: exporter-kube-etcd
    chart: exporter-kube-etcd-0.1.0
    component: kube-etcd
    heritage: Tiller
    release: kube-prometheus
  name: kube-prometheus-exporter-kube-etcd
  namespace: kube-system
  resourceVersion: "41224"
  selfLink: /api/v1/namespaces/kube-system/services/kube-prometheus-exporter-kube-etcd
  uid: 262948a4-579a-11e7-8ee6-525400f18cb3
spec:
  clusterIP: None
  ports:
  - name: http-metrics
    port: 4001
    protocol: TCP
    targetPort: 4001
  selector:
    k8s-app: etcd-server
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}

# kubectl get servicemonitors/kube-prometheus-exporter-kube-etcd -n kube-prometheus -oyaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: ServiceMonitor
metadata:
  creationTimestamp: 2017-06-22T22:28:43Z
  labels:
    app: prometheus
    chart: exporter-kube-etcd-0.1.0
    component: kube-etcd
    heritage: Tiller
    prometheus: kube-prometheus
    release: kube-prometheus
  name: kube-prometheus-exporter-kube-etcd
  namespace: kube-prometheus
  resourceVersion: "142223"
  selfLink: /apis/monitoring.coreos.com/v1alpha1/namespaces/kube-prometheus/servicemonitors/kube-prometheus-exporter-kube-etcd
  uid: 2651379c-579a-11e7-8ee6-525400f18cb3
spec:
  endpoints:
  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
    interval: 15s
    port: http-metrics
  jobLabel: kube-prometheus-exporter-kube-etcd
  namespaceSelector:
    matchNames:
    - kube-system
  selector:
    matchLabels:
      app: exporter-kube-etcd
      component: kube-etcd
```

as kubeadm defines etcd as:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/config.hash: 7075157cfd4524dbe0951e00a8e3129e
    kubernetes.io/config.mirror: 7075157cfd4524dbe0951e00a8e3129e
    kubernetes.io/config.seen: 2017-06-22T14:38:19.928172382Z
    kubernetes.io/config.source: file
  creationTimestamp: 2017-06-22T14:38:24Z
  labels:
    component: etcd
    tier: control-plane
  name: etcd-kube-master-1
  namespace: kube-system
  resourceVersion: "252"
  selfLink: /api/v1/namespaces/kube-system/pods/etcd-kube-master-1
  uid: 729e2910-5758-11e7-8ee6-525400f18cb3
```

**Comment by brancz** [comment]
FWIW monitoring etcd is not supported by kube-prometheus right now. There merely an example how it _could_ be done. The RBAC error is because those certificates are not valid for authenticating against the kubelet. Scraping secure kubelets is also not supported right now by kube-prometheus, however, we're working on getting to that, for now I recommend using the insecure port.

**Comment by gianrubio** [comment]
Removing [this default](https://github.com/coreos/prometheus-operator/blob/master/helm/prometheus/templates/prometheus.yaml#L51) `serviceMonitorSelector` can help other new users.

If the user didn't specify any `serviceMonitorSelector` we should assume the selector is just `app=prometheus` and the chart kube-prometheus will work properly without any changes.

**Comment by brancz** [comment]
@gianrubio you recently wrote a blog post and used the helm charts in it, right? Could you confirm that they are currently working when installed from the repository?

**Comment by gianrubio** [comment]
@brancz yes I wrote the post on [medium](https://itnext.io/kubernetes-monitoring-with-prometheus-in-15-minutes-8e54d1de2e13) explainning how to use prometheus-operator. The charts are not working out of the box, as you can see in the post I'm using `sed` and  specific selector.
I'm waiting #676 to be merged to have a coreos helm chart, after that I'll work to fix all this issues.

Also related to #592

**Comment by brancz** [comment]
Got it, thanks for the info! The chart repo is on our list, but as I mentioned in the PR I'd prefer to use the quay helm registry if possible, as I would like to avoid having to manage a raw s3 bucket.

**Comment by gianrubio** [comment]
@brancz can we close this? fixed on #776
